File: modules/liveview/samples/python/liveview_basic.py
import cv2 as cv
import numpy as np
import time


def main():
    started = time.time()

    def feed():
        frame = np.full((240, 320, 3), (40, 120, 220), dtype=np.uint8)
        x = int(((time.time() - started) * 80) % frame.shape[1])
        cv.rectangle(frame, (x, 80), (min(x + 60, frame.shape[1] - 1), 150),
                     (240, 240, 80), -1)
        return frame

    view = cv.liveview.show(feed, name="sample", mode="auto", show=False)
    print("LiveView URL:", view.channel_url)
    input("Press Enter to stop.")
    view.close()


if __name__ == "__main__":
    main()

File: modules/liveview/samples/cpp/liveview_basic.cpp
#include <opencv2/core.hpp>
#include <opencv2/liveview.hpp>

#include <iostream>

int main()
{
    cv::Ptr<cv::liveview::Server> view = cv::liveview::createServer("127.0.0.1", 0);
    view->start();
    view->publish("sample", cv::Mat(240, 320, CV_8UC3, cv::Scalar(40, 120, 220)));
    std::cout << "LiveView URL: " << view->url() << std::endl;
    std::cout << "Press Enter to stop." << std::endl;
    std::cin.get();
    return 0;
}
